src/gui/threejsScene.js

```diff
@@ -362,92 +376,220 @@ import { BufferGeometryUtils } from '../../thirdPartyCode/three/BufferGeometryUt
     class CustomMaterials {
         constructor() {
             this.materialsToAnimate = [];
+            this.lastUpdate = -1;
         }
-        areaTargetVertexShader() {
-            return `
-            precision highp float;
-
-            uniform float sineTime;
-            uniform float time;
-
-            uniform mat4 modelViewMatrix;
-            uniform mat4 projectionMatrix;
-
-            attribute vec3 position;
-            attribute vec4 color;
-            attribute vec3 translate;
-            attribute vec2 uv;
-
-            varying vec3 vPosition;
-            varying vec4 vColor;
-            varying float vScale;
-            varying vec2 vUv;
-
-            void main(){
-
-                vPosition = position;
-                vec3 trTime = vec3(translate.x + time,translate.y + time,translate.z + time);
-                float scale =  sin( trTime.x * 2.1 ) + sin( trTime.y * 3.2 ) + sin( trTime.z * 4.3 );
-                vScale = scale;
-
-                vColor = color;
-                vUv = uv;
-
-                gl_Position = projectionMatrix * modelViewMatrix * vec4( vPosition, 1.0 );
-
-            }
-          `
+        areaTargetVertexShader(center) {
+            return THREE.ShaderChunk.meshphysical_vert
+                .replace('#include <worldpos_vertex>', `#include <worldpos_vertex>
+    vPosition = position;
+    len = length(vPosition - vec3(${center.x}, ${center.y}, ${center.z}));
+    vWorldPosition = (modelMatrix * vec4( transformed, 1.0 )).xyz;`).replace('#include <common>', `#include <common>
+    varying vec3 vPosition;
+    varying float len;
+    varying vec3 vWorldPosition;`);
```

Comment:
Oh, I was wondering how to edit an existing three.js shader instead of starting from scratch, this is pretty neat

Reply:
Thanks! I should do this to the fragment shader as well so it's a bit more robust to changes in the THREE version but this is good enough for now